### PR TITLE
refactor(cli): enable inspector proxy with network support by default

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Flipping this switch to be opt-out for SDK 49.

# How

- Renamed `EXPO_USE_CUSTOM_INSPECTOR_PROXY=true` to `EXPO_NO_INSPECTOR_PROXY=false`

> This change made more sense as you would have to set `EXPO_USE_CUSTOM_INSPECTOR_PROXY=false` to revert back to the original (Metro) inspector proxy.

# Test Plan

- `$ EXPO_DEBUG=true expo start` -> should mention `Expo inspector proxy enabled` in the debug logs.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/more/expo-cli.mdx [hunk 1]
```diff
@@ -326,6 +326,7 @@ For example, you are testing a new version of `react-native-reanimated` to verif
 You can exclude specific packages from the version checks while still allowing the `install` command to install, check, and fix any other dependencies. This configuration extends to the checking done by `npx expo-doctor`.
 
 To exclude packages from version checking, set the `expo.install` config object in your project's **package.json**:
+
```
Comment (byCedric) [inline docs/pages/more/expo-cli.mdx:329]: I ran "format" on this page, which added this, but I can remove it 🤷 (but it seems like a good addition, separating code blocks with text a bit nicer)

File: docs/pages/more/expo-cli.mdx [hunk 2]
```diff
@@ -390,7 +391,7 @@ From here, you can choose to generate basic project files like:
...
+| `EXPO_NO_INSPECTOR_PROXY`           | **boolean** | <br/>Disable the customized inspector proxy with improved support for the Chrome DevTools protocol.<br/>This includes support for the network inspector.                                                                                                                                                                                                                             |
 | `EXPO_USE_TYPED_ROUTES`             | **boolean** | <StatusTag status="experimental" note="SDK 49+" /><br/>Generate TypeScript types for Expo Router.<br/>This includes support for the network inspector.                                                                                                                                                                                                                               |
```
Comment (byCedric) [inline docs/pages/more/expo-cli.mdx:395]: Lmao, what happened here? cc @marklawlor, I think we can drop the `This includes support for the network inspector` part from `EXPO_USE_TYPED_ROUTES` right? ([added here](https://github.com/expo/expo/pull/21651/files#diff-8c2b6f8e985e42676c1d874b0750a60dba9562499a1728eaea950dfa73e29c4eR375))

```suggestion
| `EXPO_USE_TYPED_ROUTES`             | **boolean** | <StatusTag status="experimental" note="SDK 49+" /><br/>Generate TypeScript types for Expo Router.<br/>                                                                                                                                                                                                                             |
```

File: docs/pages/more/expo-cli.mdx [hunk 3]
```diff
@@ -390,7 +391,7 @@ From here, you can choose to generate basic project files like:
...
-| `EXPO_USE_CUSTOM_INSPECTOR_PROXY`   | **boolean** | <StatusTag status="experimental" note="SDK 49+" /><br/>Use a customized inspector proxy with improved support for the Chrome DevTools protocol.<br/>This includes support for the network inspector.                                                                                                                                                                                 |
+| `EXPO_NO_INSPECTOR_PROXY`           | **boolean** | <br/>Disable the customized inspector proxy with improved support for the Chrome DevTools protocol.<br/>This includes support for the network inspector.                                                                                                                                                                                                                             |
```
Comment (Kudo) [inline docs/pages/more/expo-cli.mdx:394]: could we still keep the SDK 49+, since the guide is not versioned.

**Review by Kudo — APPROVED** [review]
leaving a nit comment. thanks for helping this 🙏

File: docs/pages/more/expo-cli.mdx (hunk 3 above)
Comment (byCedric) [reply docs/pages/more/expo-cli.mdx:394]: oh yeah, good point.

Comment (byCedric) [reply docs/pages/more/expo-cli.mdx:394]: I'll just drop the `experimental` part then :)

File: docs/ui/components/Tag/StatusTag.tsx [hunk 4]
```diff
@@ -20,10 +20,7 @@ export const StatusTag = ({ status, type, note }: StatusTagProps) => {
       )}
       css={[tagStyle, type === 'toc' && tagToCStyle]}>
       {status === 'experimental' && <Stars02Icon className="icon-xs text-palette-pink11" />}
-      <span css={labelStyle}>
-        {formatName(status)}
-        {note && `: ${note}`}
-      </span>
+      <span css={labelStyle}>{status ? formatName(status) + (note ? `: ${note}` : '') : note}</span>
```
Comment (byCedric) [inline docs/ui/components/Tag/StatusTag.tsx:23]: This was required to use the `<StatusTag />` with just `notes` (as in, for just adding "SDK XX+")

image

**Pull request merged**